core/plugins/org.polarsys.capella.core.validation/src/org/polarsys/capella/core/validation/CapellaValidationActivator.java

```diff
@@ -64,10 +62,21 @@ public static CapellaValidationActivator getDefault() {
   }
 
   /**
-   * Get the capella validator adapter.
-   * @return a not <code>null</code> instance.
+   * Returns the validator registry for capella validation. The registry is configured to
+   * funnel standard EMF validation through to EMF Constraint Validation via constraints by using an
+   * {@link CapellaValidatorAdapter} as the validator for all Capella packages and as the default
+   * for other EPackages.
    */
-  public CapellaValidatorAdapter getCapellaValidatorAdapter() {
-    return _capellaValidatorAdapter;
+  public EValidator.Registry getCapellaValidatorRegistry() {
+    return registry;
+  }
+
+  private void initializeCapellaValidatorRegistry() {
+    CapellaValidatorAdapter adapter = new CapellaValidatorAdapter(ValidationHelper.newDefaultCapellaBatchValidator());
+    for (EPackage p : CapellaPackageRegistry.getAllCapellaPackages()) {
+      registry.put(p, adapter);
+    }
+    registry.put(RePackage.eINSTANCE, adapter);
+    registry.put(null, adapter);
```

Comment:
This one is intended?

Reply:
Yes, that is intended. The diagnostician uses the registry to find an EValidator for a given Object by looking for a value for the EPackage of the Object, and if none is found, it uses the null key to find an adapter. This way, by default, extension packages (e.g. viewpoint extensions) will always use the capella adapter. All the put(ePackage...) above are only optimisations for faster lookup..